arctouch-challenge-ios/Views/ViewControllers/UpcomingMoviesViewController.swift

```diff
@@ -0,0 +1,145 @@
+    @IBOutlet weak var searchBar: UISearchBar!
+    
+    var upcomingMoviesPage: Int = 1
+    var selectedMovie: Movie?
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        self.searchBar.delegate = self
+        self.setupCollectionView()
+        self.setupNavigationBar()
+        self.viewModel = UpcomingViewModel()
+    }
+
+    private func setupNavigationBar() {
+        title = NSLocalizedString("Upcoming Movies",
+                                       comment: "Title for navigation bar on UpcomingMovies List")
+    }
+    
+    private func setupCollectionView() {
+        collectionView.delegate = nil
+        collectionView.dataSource = nil
+        collectionView.register(UINib(nibName: "MovieCollectionViewCell", bundle: nil),
+                                forCellWithReuseIdentifier: MovieCollectionViewCell.identifier)
+    }
+    
+    override func didReceiveMemoryWarning() {
+        super.didReceiveMemoryWarning()
+        // Dispose of any resources that can be recreated.
+    }
+    
+    func fetchMovies(for page: Int) {
+        viewModel?.fetchUpcomingMovies(for: upcomingMoviesPage)
+    }
+    
+    override func setupBindings() {
+        viewModel?.upcomingMovies.bind(to: collectionView.rx.items(cellIdentifier: MovieCollectionViewCell.identifier,
+                                                                   cellType: MovieCollectionViewCell.self)) {  row, model, cell in
+                                                                    cell.presenter = MoviePresenter(movie: model)
+                                                                    
+                                                                    if let numberOfMovies = self.viewModel?.moviesCount() {
+                                                                        if row >= (numberOfMovies - 1) {
+                                                                            self.upcomingMoviesPage += 1
+                                                                            self.fetchMovies(for: self.upcomingMoviesPage)
+                                                                        }
+                                                                    }
+            }
+            .disposed(by: disposeBag)
+        
+        collectionView.rx.itemSelected.subscribe(onNext: { [weak self] indexPath in
+            guard let selectedMovie = self?.viewModel?.upcomingMovies.value[indexPath.row] else {
+                return
+            }
+            self?.collectionView.deselectItem(at: indexPath, animated: true)
+            self?.selectedMovie = selectedMovie
+            self?.performSegue(withIdentifier: "showDetails", sender: self)
+        }).disposed(by: disposeBag)
+        
+        searchBar.rx.text
+            .orEmpty
+            .debounce(0.5, scheduler: MainScheduler.instance)
+            .asDriver(onErrorJustReturn: "")
+            .drive((viewModel?.searchText)!)
```

Comment:
I noticed that you use force unwrap quite 
Is this a pattern that you always follow?
If not, how would you handle a crash in this situation?

Reply:
I only do force unwrap in situations I'm sure it won't ever be nil, in the commented part, for instance, I'm always sure that viewModel and searchText will always be instantiated when this method is called.
Whenever I can't be sure (dealing with objects that come from a server, for instance) I use if let or guard let, to make sure.